### DISCUSSION TITLE
UDMSE no "dev iptv" is shown | HK isp - hkt netvigator, iptv - now tv

### DISCUSSION THREAD

**Discussion by izenabizenak** [fabianishere/udm-iptv#95] [Q&A]
Goedemorgen Fabian. Thank you for creating this project to help udm users.

Been new to linux/debien and UDMSE (2.3.15.6554), I have installed / uninstalled the code a few times since yesterday 2022-03-11. no success so far. 

issues:
### .met the same warning as Qlogic711. 
Created symlink /etc/systemd/system/multi-user.target.wants/udm-iptv.service -> /lib/systemd/system/udm-iptv.service.
N: Download is performed unsandboxed as root as file '/root/igmpproxy_0.3-1_arm64.deb' couldn't be accessed by user '_apt'. - pkgAcquire::Run (13: Permission denied)

### . per below, apt cannot updated? and debconf: unable to initialize frontend: Dialog?
root@UDMSE:~# apt update && apt install dialog
Get:1 http://security.debian.org stretch/updates InRelease [53.0 kB]
Ign:2 http://deb.debian.org/debian stretch InRelease
Hit:4 http://deb.debian.org/debian stretch-updates InRelease
Get:5 http://security.debian.org stretch/updates/main arm64 Packages [739 kB]
Get:6 http://deb.debian.org/debian stretch-backports InRelease [91.8 kB]
Hit:7 http://deb.debian.org/debian stretch Release
Hit:8 https://apt-beta.artifacts.ui.com stretch InRelease
Hit:9 https://apt-release-candidate.artifacts.ui.com stretch InRelease
Hit:10 https://apt.artifacts.ui.com stretch InRelease
Hit:3 https://dl.ubnt.com/unifi/debian cloudkey-stable InRelease
Fetched 884 kB in 2s (393 kB/s)
Reading package lists... Done
Building dependency tree
Reading state information... Done
16 packages can be upgraded. Run 'apt list --upgradable' to see them.
Reading package lists... Done
Building dependency tree
Reading state information... Done
The following NEW packages will be installed:
  dialog
0 upgraded, 1 newly installed, 0 to remove and **16 not upgraded**.
Need to get 243 kB of archives.
After this operation, 1084 kB of additional disk space will be used.
Get:1 http://deb.debian.org/debian stretch/main arm64 dialog arm64 1.3-20160828-2 [243 kB]
Fetched 243 kB in 0s (887 kB/s)
**debconf: unable to initialize frontend: Dialog**
debconf: (No usable dialog-like program is installed, so the dialog based frontend cannot be used. at /usr/share/perl5/Debconf/FrontEnd/Dialog.pm line 76, <> line 1.)
debconf: falling back to frontend: Readline
Selecting previously unselected package dialog.
(Reading database ... 73263 files and directories currently installed.)
Preparing to unpack .../dialog_1.3-20160828-2_arm64.deb ...
Unpacking dialog (1.3-20160828-2) ...
Setting up dialog (1.3-20160828-2) ...

### .16 not upgraded?
root@UDMSE:~# apt install ./igmpproxy_0.3-1_arm64.deb ./udm-iptv_2.1.0_all.deb
Reading package lists... Done
Building dependency tree
Reading state information... Done
Note, selecting 'igmpproxy' instead of './igmpproxy_0.3-1_arm64.deb'
Note, selecting 'udm-iptv' instead of './udm-iptv_2.1.0_all.deb'
The following NEW packages will be installed:
  igmpproxy udm-iptv
0 upgraded, 2 newly installed, 0 to remove and **16 not upgraded**.
Need to get 0 B/53.7 kB of archives.
After this operation, 158 kB of additional disk space will be used.
Get:1 /root/igmpproxy_0.3-1_arm64.deb igmpproxy arm64 0.3-1 [38.3 kB]
Get:2 /root/udm-iptv_2.1.0_all.deb udm-iptv all 2.1.0 [15.3 kB]
Preconfiguring packages ...

### . no separate IPTV VLAN is presumed 
Since this IPTV passthru worked with Asus RT-AX88U router configured with "ISP Profile" as "None", I presume this IPTV setting is of generic and hence of no separate VLAN for IPTV. I used VLAN 0 in the conf file in UDMSE 
![20220312105420](https://user-images.githubusercontent.com/66798734/158002013-d9dda5bb-2dde-423c-92aa-c6782a0ecf97.png)

[udm-iptv.conf.txt](https://github.com/fabianishere/udm-iptv/files/8236975/udm-iptv.conf.txt)

I tried both default range "213.75.0.0/16,217.166.0.0/16  " and “0.0.0.0/0” but no success. I am asking more details of parameter from the ISP. Hope they would disclose to me.

root@UDMSE:~# ip -4 addr show dev iptv
root@UDMSE:~# ip route list
10.18.20.1 dev tun2 proto kernel scope link src 10.18.20.2
10.20.120.2 dev tun1 proto kernel scope link src 10.20.120.1
42.98.xxx.0/24 dev eth8 proto kernel scope link src 42.98.xxx.16
192.168.18.0/24 via 10.18.20.1 dev tun2
192.168.20.0/24 dev br0 proto kernel scope link src 192.168.20.1
192.168.22.0/24 dev br22 proto kernel scope link src 192.168.22.1
192.168.23.0/24 dev br23 proto kernel scope link src 192.168.23.1
192.168.24.0/24 dev br24 proto kernel scope link src 192.168.24.1
192.168.25.0/24 dev br25 proto kernel scope link src 192.168.25.1
192.168.120.0/22 via 10.20.120.2 dev tun1
192.168.124.0/23 via 10.20.120.2 dev tun1
root@UDMSE:~# systemctl restart udm-iptv
root@UDMSE:~# udm-iptv-diag

Please share the following output with the developers:
=== Configuration ===
WAN Interface: eth8
WAN Ranges: 213.75.0.0/16 217.166.0.0/16
WAN VLAN: 0 (dev iptv)
WAN DHCP options: -O staticroutes -V IPTV_RG
LAN Interfaces: br0  br23
IGMP Proxy quickleave disabled: false
IGMP Proxy debug: true
=== IP Link and Route ===
11: eth8: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc htb state UP group default qlen 1000
    inet 42.98.xxx.yyy/24 scope global dynamic eth8
       valid_lft 1792sec preferred_lft 1792sec
42.98.xxx.0/24 proto kernel scope link src 42.98.xxx.yyy
=== Service Logs ===
Mar 12 10:47:18 UDMSE udm-iptv[394]: RECV Membership query   from 192.168.20.1    to 224.0.0.1
Mar 12 10:47:18 UDMSE udm-iptv[394]: RECV Membership query   from 192.168.23.1    to 224.0.0.1
Mar 12 10:47:18 UDMSE udm-iptv[394]: RECV V2 member report   from 169.254.52.174  to 224.0.0.1
Mar 12 10:47:18 UDMSE udm-iptv[394]: No interfaces found for source 169.254.52.174
Mar 12 10:47:18 UDMSE udm-iptv[394]: The IGMP message was local multicast. Ignoring.
Mar 12 10:47:19 UDMSE udm-iptv[394]: RECV V2 member report   from 192.168.20.3    to 233.89.188.1
Mar 12 10:47:19 UDMSE udm-iptv[394]: Updated route entry for 233.89.188.1 on VIF #1
Mar 12 10:47:19 UDMSE udm-iptv[394]: The source address 192.168.20.3 for group 233.89.188.1 is from downstream VIF[0]. Ignoring.
Mar 12 10:47:19 UDMSE udm-iptv[394]: RECV V2 member report   from 192.168.20.1    to 224.0.0.2
Mar 12 10:47:19 UDMSE udm-iptv[394]: The IGMP message was from myself. Ignoring.
root@UDMSE:~# dpkg-reconfigure -p medium udm-iptv

root@UDMSE:~# systemctl restart udm-iptv
root@UDMSE:~# udm-iptv-diag
Please share the following output with the developers:
=== Configuration ===
WAN Interface: eth8
WAN Ranges: 0.0.0.0/0
WAN VLAN: 0 (dev iptv)
WAN DHCP options: -O staticroutes -V IPTV_RG
LAN Interfaces: br23
IGMP Proxy quickleave disabled: false
IGMP Proxy debug: true
=== IP Link and Route ===
11: eth8: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc htb state UP group default qlen 1000
    inet 42.98.xxx.yyy/24 scope global dynamic eth8
       valid_lft 1688sec preferred_lft 1688sec
42.98.xxx.0/24 proto kernel scope link src 42.98.xxx.yyy
=== Service Logs ===
Mar 12 10:48:55 UDMSE udm-iptv[10014]: The IGMP message was from myself. Ignoring.
Mar 12 10:48:55 UDMSE udm-iptv[10014]: RECV V2 member report   from 192.168.23.1    to 224.0.0.22
Mar 12 10:48:55 UDMSE udm-iptv[10014]: The IGMP message was from myself. Ignoring.
Mar 12 10:48:56 UDMSE udm-iptv[10014]: The IGMP message was local multicast. Ignoring.
Mar 12 10:48:56 UDMSE udm-iptv[10014]: RECV V2 member report   from 192.168.23.1    to 233.89.188.1
Mar 12 10:48:56 UDMSE udm-iptv[10014]: The IGMP message was from myself. Ignoring.
Mar 12 10:48:56 UDMSE udm-iptv[10014]: Inserted route table entry for 233.89.188.1 on VIF #-1
Mar 12 10:48:59 UDMSE udm-iptv[10014]: Route activation request from 42.98.xxx.yyy for 233.89.188.1 is from myself. Ignoring.
Mar 12 10:49:02 UDMSE udm-iptv[10014]: RECV V2 member report   from 192.168.23.1    to 224.0.0.251
Mar 12 10:49:02 UDMSE udm-iptv[10014]: The IGMP message was from myself. Ignoring.

Thanks for your help in advance.

**Comment by fabianishere** [comment]
You can ignore the first three warnings. Looking at your ASUS router configuration, it seems like you are using bridged IPTV as opposed to routed IPTV (which is what this project does).

I recommend asking your ISP whether they support routed IPTV and what addresses they use for it.
